atrium-api/src/app/bsky/embed/record.rs

```diff
@@ -46,7 +46,7 @@ pub struct ViewRecordData {
     pub repost_count: Option<i64>,
     pub uri: String,
     ///The record data itself.
-    pub value: crate::records::Record,
+    pub value: crate::types::Unknown,
```

Comment:
Hmm, the `moderation_quoteposts` test now fails to compile because it expects a known record type here. Scanning over the other changes, it does seem that _currently_ an `unknown` field named `value` is consistently a `Record`, but then why do the lexicons not use an open `union`?

I think we're going to need to ask for further guidance here. It might be that what we need to do, instead of my current changes, is to make `UnknownData.type` an `Option<String>` and then enforce that it is `Some(_)` everywhere that it is used for a `union` field.

Reply:
I've commented about the issue here: https://github.com/bluesky-social/atproto/issues/999#issuecomment-2254334072